pkg/api/api.go

```diff
@@ -63,11 +63,17 @@ type Config struct {
 	// initialized, the custom config handler will be used instead of
 	// DefaultConfigHandler.
 	CustomConfigHandler ConfigHandler `yaml:"-"`
+
+	// These allow and are used to configure the addition of HTTP Header fields to logs
+	LogHeaders           bool   `yaml:"LogHeaders"`
+	TargetRequestHeaders string `yaml:"TargetRequestHeaders"`
 }
 
 // RegisterFlags adds the flags required to config this to the given FlagSet.
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.ResponseCompression, "api.response-compression-enabled", false, "Use GZIP compression for API responses. Some endpoints serve large YAML or JSON blobs which can benefit from compression.")
+	f.BoolVar(&cfg.LogHeaders, "api.LogHeaders", false, "Enable logging of header specific context information")
+	f.StringVar(&cfg.TargetRequestHeaders, "api.TargetRequestHeaders", "", "Target Headers for Request logging (if enabled) Separate multiple headers with ~, whitespace is ignored")
```

Comment:
Hmm i think cortex has another default for list of strings instead of ~ separator.
I was looking at 
`[forward_headers_list: <list of string> | default = []]`
and in the code they are using `flagext.StringSlice`
cant we also use that?

Reply:
StringSlice has some sort of issue that I couldn't figure out with (un)marshalling the yaml. However, I saw there was a `flagext.StringSliceCSV` that ended up working well. I've adjusted it in the new commit.